analyses/malawi/flooding_trigger/mwi-glofas-skill.md

```diff
@@ -0,0 +1,231 @@
+
+import read_in_data as rd
+from importlib import reload
+reload(rd)
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+
+from src.indicators.flooding.config import Config
+
+config = Config()
+mpl.rcParams['figure.dpi'] = 300
+
+PLOT_DIR = config.DATA_DIR / 'processed' / 'mwi' / 'plots' / 'flooding'
+EXPLORE_DIR = config.DATA_DIR / 'exploration' / 'mwi' / 'flooding'
+GLOFAS_VERSION = 3
+STATIONS = ['glofas_1', 'glofas_2']
+SAVE_FIG = True
+```
+
+### Read in the GloFAS data
+
+```python
+da_glofas_reanalysis = {}
+da_glofas_reforecast = {}
+da_glofas_forecast = {}
+da_glofas_forecast_summary = {}
+da_glofas_reforecast_summary = {}
+
+for station in STATIONS: 
+    da_glofas_reanalysis[station] = rd.get_glofas_reanalysis(version=GLOFAS_VERSION, station=station)
+    da_glofas_reforecast[station] = rd.get_glofas_reforecast(version=GLOFAS_VERSION, station=station)
+    da_glofas_forecast[station] = rd.get_glofas_forecast(version=GLOFAS_VERSION, station=station)
+    da_glofas_forecast_summary[station] = rd.get_da_glofas_summary(da_glofas_forecast[station])
+    da_glofas_reforecast_summary[station] = rd.get_da_glofas_summary(da_glofas_reforecast[station])
+```
+
+### Read in the return period thresholds
+
+```python
+df_rps = pd.read_csv(EXPLORE_DIR / 'glofas_rps.csv')
+rp_thresh = [2, 3, 5] # The rp thresholds that we're concerned about checking
+skill_thresh_vals = {}
+for station in STATIONS:
+    skill_thresh_vals[station] = [df_rps[df_rps['rp']==thresh][station].iloc[0] for thresh in rp_thresh]
+```
+
+### Calculate the skill across lead times
+
+```python
+def is_rainy_season(month):
+    # Oct to April
+    return (month >= 10) | (month <= 4)
+
+def is_dry_season(month):
+    # May to Sept
+    return (month < 10) & (month > 4)
+
+def plot_skill(df_crps, division_key=None, add_line_from_website=False,
+              ylabel="CRPS [m$^3$ s$^{-1}$]"):
+    fig, ax = plt.subplots()
+    for station, ls in zip(STATIONS, [':', '--']):
+        df = df_crps[station].copy()
+        for i, subset in enumerate([None] + skill_thresh_vals[station]):
+            ykey = f'crps_{subset}' if subset is not None else 'crps'
+            y = df[ykey]
+            if division_key is not None:
+                dkey = f'{division_key}_{subset}' if subset is not None else division_key
+                y /= df[dkey]
+            ax.plot(df['leadtime'], y, ls=ls, c=f'C{i}')
+        ax.plot([], [], ls=ls, c='k', label=f'{station}')
+    if add_line_from_website:
+        ax.plot(df_skill[0], df_skill[1], ls='-', c='k', lw=0.5, label='from website')
+    # Add colours to legend
+    for i, subset in enumerate(['full year'] + [f'{thresh}-year return period' for thresh in rp_thresh]):
+        ax.plot([], [], c=f'C{i}', label=subset)
+    ax.set_xlabel("Lead time (days)")
+    ax.set_ylabel(ylabel)
+    ax.legend()
+```
+
+```python
+df_crps = {
+    STATIONS[0]: pd.DataFrame(columns=['leadtime', 'crps']),
```

Comment:
Small thing, but you can move this line to right under `for station in STATIONS`, then you don't have to repeat it for each station